cmake/ccache.cmake

```diff
@@ -0,0 +1,43 @@
+# Adapted from https://crascit.com/2016/04/09/using-ccache-with-cmake/
+cmake_minimum_required(VERSION 3.4)
+
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+  # Set up wrapper scripts
+  set(C_LAUNCHER   "${CCACHE_PROGRAM}")
+  set(CXX_LAUNCHER "${CCACHE_PROGRAM}")
+
+  file(WRITE "${CMAKE_BINARY_DIR}/launch-c" ""
```

Comment:
The `launch-c` etc. hack is not necessary as we only need/want this working for Makefiles and Ninja, which have official support in CMake for the `LAUNCHER` method.